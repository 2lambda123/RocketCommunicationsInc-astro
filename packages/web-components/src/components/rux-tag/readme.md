# rux-tag



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description                                        | Type                                        | Default     |
| -------- | --------- | -------------------------------------------------- | ------------------------------------------- | ----------- |
| `status` | `status`  | Used to display a status of pass, fail, or unkown. | `"fail" \| "pass" \| "unkown" \| undefined` | `undefined` |


## Shadow Parts

| Part          | Description |
| ------------- | ----------- |
| `"container"` |             |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
