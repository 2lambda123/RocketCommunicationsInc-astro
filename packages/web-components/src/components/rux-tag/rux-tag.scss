:host {
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: var(--radius-base);
    font-size: var(--font-body-2-bold-font-size);
    font-weight: var(--font-body-2-bold-font-weight);
    line-height: var(--font-body-2-bold-line-height);
    font-family: var(--font-body-2-bold-font-family);
    letter-spacing: var(--font-body-2-bold-letter-spacing);
    padding: 0.25rem 0.625rem;
    text-align: center;
}
:host(.is-undefined) {
    height: 1.5rem;
    width: 3.75rem;
    border: 1px solid #9ea7ad;
    background: #32363a;
}

:host([status='pass']) {
    height: 1.5rem;
    width: 3.75rem;
    border: 1px solid #56f000; //dark/status/normal-fill
    background: #015800; //global/tag/status/pass
}

:host([status='fail']) {
    height: 1.5rem;
    width: 3.75rem;
    border: 1px solid #ff3838;
    background: #570d01;
}

:host([status='unknown']) {
    height: 1.5rem;
    width: 3.75rem;
    border: 1px solid #9ea7ad;
    background: #32363a;
}
