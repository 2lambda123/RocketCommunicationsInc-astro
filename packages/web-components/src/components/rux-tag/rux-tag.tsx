import { Component, Host, h, Prop, Element, State } from '@stencil/core'
import { StatusTags } from '../../common/commonTypes.module'
import { hasSlot } from '../../utils/utils'

const statusMap = {
    unknown: 'UNK',
    pass: 'PASS',
    fail: 'FAIL',
}

/**
 * @slot (default) - The text for the rux-tag
 * @part container - The container of the rux-tag's text
 */

@Component({
    tag: 'rux-tag',
    styleUrl: 'rux-tag.scss',
    shadow: true,
})
export class RuxTag {
    /**
     *  Used to display a status of pass, fail, or unknown. If no status is provided or the provided status is not an accepted status type, the default is unknown.
     */
    @Prop({ reflect: true }) status?: StatusTags = 'unknown'

    @Element() el!: HTMLRuxTagElement

    @State() hasSlot: boolean = false

    connectedCallback() {
        this._handleSlotChange = this._handleSlotChange.bind(this)
        this.hasSlot = hasSlot(this.el)
    }

    private _handleSlotChange() {
        this.hasSlot = hasSlot(this.el)
    }

    private _getValidStatus() {
        if (this.status) {
            //if it is a valid status, return it
            if (statusMap[this.status]) {
                return statusMap[this.status]
            }
            //if it's not, return the default status
            else {
                return statusMap['unknown']
            }
        }
    }

    render() {
        return (
            <Host
                class={{
                    'is-undefined': statusMap[this.status!] === undefined,
                }}
            >
                <div part="container">
                    <slot onSlotchange={this._handleSlotChange}></slot>
                    {!this.hasSlot ? this._getValidStatus() : null}
                </div>
            </Host>
        )
    }
}
