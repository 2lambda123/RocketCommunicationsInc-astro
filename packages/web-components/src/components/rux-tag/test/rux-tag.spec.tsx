import { newSpecPage } from '@stencil/core/testing'
import { RuxTag } from '../rux-tag'

describe('rux-tag', () => {
    it('renders', async () => {
        const page = await newSpecPage({
            components: [RuxTag],
            html: `<rux-tag></rux-tag>`,
        })
        expect(page.root).toMatchSnapshot()
    })
    it('renders pass status', async () => {
        const page = await newSpecPage({
            components: [RuxTag],
            html: `<rux-tag status="pass"><rux-tag>`,
        })
        expect(page.root).toMatchSnapshot()
    })
    it('renders fail status', async () => {
        const page = await newSpecPage({
            components: [RuxTag],
            html: `<rux-tag status="fail"><rux-tag>`,
        })
        expect(page.root).toMatchSnapshot()
    })
    it('renders unk status', async () => {
        const page = await newSpecPage({
            components: [RuxTag],
            html: `<rux-tag status="unknown"><rux-tag>`,
        })
        expect(page.root).toMatchSnapshot()
    })
    it('renders default unk status if status is invalid', async () => {
        const page = await newSpecPage({
            components: [RuxTag],
            html: `<rux-tag status="bad"></rux-tag>`,
        })
        const tag = document.querySelector('rux-tag')
        expect(tag!.shadowRoot!.children[0].innerHTML).toEqual('UNK')
        expect(page.root).toMatchSnapshot()
    })
    it('renders tag-text slot', async () => {
        const page = await newSpecPage({
            components: [RuxTag],
            html: `<rux-tag>
          <div>Custom Text</div>
        </rux-tag>`,
        })
        const tag = document.querySelector('rux-tag div')
        expect(tag?.innerHTML).toEqual('Custom Text')
        expect(page.root).toMatchSnapshot()
    })
})
