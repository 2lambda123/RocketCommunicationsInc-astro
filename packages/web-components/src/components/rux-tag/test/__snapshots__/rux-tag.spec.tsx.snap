// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rux-tag renders 1`] = `
<rux-tag status="unknown">
  <mock:shadow-root>
    <div part="container">
      UNK
    </div>
  </mock:shadow-root>
</rux-tag>
`;

exports[`rux-tag renders default unk status if status is invalid 1`] = `
<rux-tag class="is-undefined" status="bad">
  <mock:shadow-root>
    <div part="container">
      UNK
    </div>
  </mock:shadow-root>
</rux-tag>
`;

exports[`rux-tag renders fail status 1`] = `
<rux-tag status="fail">
  <mock:shadow-root>
    <div part="container">
      FAIL
    </div>
  </mock:shadow-root>
  <rux-tag status="unknown">
    <mock:shadow-root>
      <div part="container">
        UNK
      </div>
    </mock:shadow-root>
  </rux-tag>
</rux-tag>
`;

exports[`rux-tag renders pass status 1`] = `
<rux-tag status="pass">
  <mock:shadow-root>
    <div part="container">
      PASS
    </div>
  </mock:shadow-root>
  <rux-tag status="unknown">
    <mock:shadow-root>
      <div part="container">
        UNK
      </div>
    </mock:shadow-root>
  </rux-tag>
</rux-tag>
`;

exports[`rux-tag renders slot 1`] = `
<rux-tag status="unknown">
  <mock:shadow-root>
    <div part="container">
      UNK
    </div>
  </mock:shadow-root>
  Custom Text
</rux-tag>
`;

exports[`rux-tag renders unk status 1`] = `
<rux-tag status="unknown">
  <mock:shadow-root>
    <div part="container">
      UNK
    </div>
  </mock:shadow-root>
  <rux-tag status="unknown">
    <mock:shadow-root>
      <div part="container">
        UNK
      </div>
    </mock:shadow-root>
  </rux-tag>
</rux-tag>
`;
